code/modules/spells/spell_types/rod_form.dm

```diff
@@ -12,47 +12,86 @@
 	invocation_type = INVOCATION_SHOUT
 	action_icon_state = "immrod"
 
-/obj/effect/proc_holder/spell/targeted/rod_form/cast(list/targets,mob/user = usr)
-	var/area/A = get_area(user)
-	if(istype(A, /area/wizard_station))
+/obj/effect/proc_holder/spell/targeted/rod_form/cast(list/targets, mob/user = usr)
+	var/area/our_area = get_area(user)
+	if(istype(our_area, /area/wizard_station))
 		to_chat(user, span_warning("You know better than to trash Wizard Federation property. Best wait until you leave to use [src]."))
 		return
-	for(var/mob/living/M in targets)
-		var/turf/start = get_turf(M)
-		var/obj/effect/immovablerod/wizard/W = new(start, get_ranged_target_turf(start, M.dir, (15 + spell_level * 3)))
-		W.wizard = M
-		W.max_distance += spell_level * 3 //You travel farther when you upgrade the spell
-		W.damage_bonus += spell_level * 20 //You do more damage when you upgrade the spell
-		W.start_turf = start
-		M.forceMove(W)
-		M.notransform = TRUE
-		M.status_flags |= GODMODE
-
-//Wizard Version of the Immovable Rod
+	for(var/mob/living/caster in targets)
+		var/turf/start = get_turf(caster)
+		var/obj/effect/immovablerod/wizard/wiz_rod = new(start, get_ranged_target_turf(start, caster.dir, (15 + spell_level * 3)))
+		wiz_rod.our_wizard = WEAKREF(caster)
+		wiz_rod.max_distance += spell_level * 3 //You travel farther when you upgrade the spell
+		wiz_rod.damage_bonus += spell_level * 20 //You do more damage when you upgrade the spell
+		wiz_rod.start_turf = start
```

Comment:
Any chance to calculate these values outside of the `for` loop, before it starts, and then to add them as initialization parameters instead of modifying the rod after spawning?